extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -82,6 +86,58 @@ describe('Project: sqlproj content operations', function (): void {
 
 		await testUtils.shouldThrowSpecificError(async () => await project.addToProject(list), `ENOENT: no such file or directory, stat \'${nonexistentFile}\'`);
 	});
+
+	it('Should add dacpac references to sqlproj', async function (): Promise<void> {
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newProjectFileBaseline);
+		const project = new Project(projFilePath);
+		await project.readProjFile();
+
+		await project.addMasterDatabaseReference();
+		await project.addDatabaseReference(Uri.parse('testPath/testSameDb.dacpac'), DatabaseReferenceLocation.sameDatabase);
+		await project.addDatabaseReference(Uri.parse('testPath/testDifferentDbSameServer.dacpac'), DatabaseReferenceLocation.differentDatabaseSameServer, 'testDb');
+
+		let projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText).equal(baselines.dacpacReferencesProjectFileBaseline.trim());
+	});
+
+	it('Should choose correct master dacpac', async function(): Promise<void> {
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newProjectFileBaseline);
+		const project = new Project(projFilePath);
+		await project.readProjFile();
+
+		if (isWindows) {
+			let uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '\\$(NETCoreTargetsPath)\\SystemDacpacs\\130\\master.dacpac');
+
+			project.changeDSP(TargetPlatform.Sql150.toString());
+			uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '\\$(NETCoreTargetsPath)\\SystemDacpacs\\150\\master.dacpac');
+
+			project.changeDSP(TargetPlatform.SqlAzureV12.toString());
+			uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '\\$(NETCoreTargetsPath)\\SystemDacpacs\\AzureV12\\master.dacpac');
+		} else {
+			let uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '/$(NETCoreTargetsPath)/SystemDacpacs/130/master.dacpac');
+
+			project.changeDSP(TargetPlatform.Sql150.toString());
+			uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '/$(NETCoreTargetsPath)/SystemDacpacs/150/master.dacpac');
+
+			project.changeDSP(TargetPlatform.SqlAzureV12.toString());
+			uri = project.getMasterDacpac();
+			should.equal(uri.fsPath, '/$(NETCoreTargetsPath)/SystemDacpacs/AzureV12/master.dacpac');
+		}
```

Comment:
Can we instead use path.join to get platform independent path location (and then verifying it again uri.fsPath)?